frontend/src/metabase/query_builder/components/filters/BulkFilterModal/BulkFilterModal.tsx

```diff
@@ -91,6 +91,11 @@ export function BulkFilterModal({
     [initialQuery, query],
   );
 
+  const handleClose = useCallback(() => {
```

Comment:
This is a controlled component, so it'd be more appropriate to add an effect that resets unsaved changes when `opened` prop changes.